src/platform/backends/shared/base_virtual_machine.cpp

```diff
@@ -277,58 +228,82 @@ void mp::BaseVirtualMachine::renew_ssh_session()
 {
     {
         const std::unique_lock lock{state_mutex};
-        if (!MP_UTILS.is_running(
-                current_state())) // spend time updating state only if we need a new session
+        if (!MP_UTILS.is_running(current_state())) // avoid wasting time
             throw SSHVMNotRunning{"SSH unavailable on instance {}: not running", vm_name};
     }
 
     mpl::debug(vm_name, "{} SSH session", ssh_session ? "Renewing cached" : "Caching new");
-
     ssh_session.emplace(ssh_hostname(), ssh_port(), ssh_username(), key_provider);
 }
+bool multipass::BaseVirtualMachine::unplugged()
+{
+    auto st = current_state();
+    return st == State::off || st == State::stopped;
+}
+
+void mp::BaseVirtualMachine::detect_aborted_start()
+{
+    const std::lock_guard lock{state_mutex};
```

Comment:
`const` is redundant here

Reply:
Yeah, I just saw that too:

```c++
const std::lock_guard<std::mutex> lock(g_i_mutex);
```
I mean, it wouldn't hurt, but it practically changes no behavior.  I'm fine with it both ways.